packages/mds-policy-service/@types/intents/index.ts

```diff
@@ -2,61 +2,120 @@ import type { BaseRule } from '../models'
 }
 
 export const INTENT_RULE_CONSTANTS: { [K in INTENT_TYPE]: IntentRuleDefaults } = {
   no_parking: NO_PARKING_INTENT_RULE_CONSTANTS,
-  permitted_vehicle_count: PERMITTED_VEHICLE_COUNT_INTENT_RULE_CONSTANTS
+  permitted_vehicle_count: PERMITTED_VEHICLE_COUNT_INTENT_RULE_CONSTANTS,
+  parking_time_limit: PARKING_TIME_LIMIT_INTENT_RULE_CONSTANTS,
+  required_parking: REQUIRED_PARKING_INTENT_RULE_CONSTANTS
 }
 
-export type IntentDraft<
-  I extends INTENT_TYPE,
-  INTENT_POLICY_FIELDS extends BaseIntentPolicyUserFields = NoParkingIntentPolicyUserFields,
-  INTENT_RULE_FIELDS extends BaseIntentRuleUserFields = NoParkingIntentRuleUserFields
-> = {
+export type IntentDraft<I extends INTENT_TYPE> = {
   intent_type: I
-  policy_fields: INTENT_POLICY_FIELDS
-  rule_fields: INTENT_RULE_FIELDS
+  policy_fields: BaseIntentPolicyUserFields
+  rule_fields: I extends 'no_parking'
+    ? NoParkingIntentRuleUserFields
+    : I extends 'permitted_vehicle_count'
+    ? PermittedVehicleCountIntentRuleUserFields
+    : I extends 'parking_time_limit'
+    ? ParkingTimeLimitIntentRuleUserFields
+    : I extends 'required_parking'
+    ? RequiredParkingIntentRuleUserFields
+    : NoParkingIntentRuleUserFields
 }
 
-export type BaseIntentRuleUserFields = NoParkingIntentRuleUserFields | PermittedVehicleCountIntentRuleUserFields
+export type BaseIntentRuleUserFields =
+  | NoParkingIntentRuleUserFields
+  | PermittedVehicleCountIntentRuleUserFields
+  | ParkingTimeLimitIntentRuleUserFields
+  | RequiredParkingIntentRuleUserFields
 
-export type NoParkingIntentRuleUserFields = Pick<
-  BaseRule<PartialMicroMobilityStatesMap>,
-  'geographies' | 'days' | 'start_time' | 'end_time' | 'vehicle_types' | 'propulsion_types'
->
 export type NoParkingIntentPolicyUserFields = BaseIntentPolicyUserFields
+
+export type NoParkingIntentRuleUserFields = [
+  Pick<
+    BaseRule<PartialMicroMobilityStatesMap>,
+    'geographies' | 'days' | 'start_time' | 'end_time' | 'vehicle_types' | 'propulsion_types'
+  >
+]
+
+export type NoParkingIntentDraft = IntentDraft<'no_parking'>
+
 export type PermittedVehicleCountIntentPolicyUserFields = BaseIntentPolicyUserFields
 
-export type NoParkingIntentDraft = IntentDraft<
-  'no_parking',
-  NoParkingIntentPolicyUserFields,
-  NoParkingIntentRuleUserFields
->
+export type PermittedVehicleCountIntentRuleUserFields = [
+  Pick<
+    BaseRule<PartialMicroMobilityStatesMap>,
+    'minimum' | 'maximum' | 'geographies' | 'days' | 'start_time' | 'end_time' | 'vehicle_types' | 'propulsion_types'
+  >
+]
+
+export type PermittedVehicleCountIntentDraft = IntentDraft<'permitted_vehicle_count'>
+
+export type ParkingTimeLimitIntentPolicyUserFields = BaseIntentPolicyUserFields
 
-export type PermittedVehicleCountIntentRuleUserFields = Pick<
+export type ParkingTimeLimitIntentRuleUserFields = [
+  Pick<
+    BaseRule<PartialMicroMobilityStatesMap>,
+    'maximum' | 'geographies' | 'days' | 'start_time' | 'end_time' | 'vehicle_types' | 'propulsion_types'
```

Comment:
I think we also need the `rule_units` field as well since that's an option in the UI